docs/how-to-guides/rook-ceph-storage.md

```diff
@@ -0,0 +1,291 @@
+- [Prerequisites](#prerequisites)
+- [Deploy storage worker pool](#deploy-storage-worker-pool)
+  - [Config](#config)
+  - [Deploy the worker pool](#deploy-the-worker-pool)
+- [Deploy Rook](#deploy-rook)
+  - [Config](#config-1)
+  - [Deploy the component](#deploy-the-component)
+- [Deploy Rook Ceph](#deploy-rook-ceph)
+  - [Config](#config-2)
+  - [Deploy the component](#deploy-the-component-1)
+- [Access the Ceph dashboard](#access-the-ceph-dashboard)
+- [Enable and access toolbox](#enable-and-access-toolbox)
+- [Enable monitoring](#enable-monitoring)
+- [Make default storage class](#make-default-storage-class)
+- [Additional resources](#additional-resources)
+
+## Introduction
+
+This guide provides the steps for deploying a storage stack using the `rook` and `rook-ceph` Lokomotive component and explains how to access Ceph dashboard, Ceph toolbox and how to enable monitoring.
+
+At the end of this tutorial you will have a storage backed by `rook` and `rook-ceph`. Any application in the cluster can request storage using Kubernetes PVC.
+
+## Prerequisites
+
+* A Lokomotive cluster deployed on a Packet cloud and accessible via `kubectl`.
+
+## Deploy storage worker pool
+
+#### Config
+
+Deploy a cluster with at least one worker pool dedicated to `rook-ceph`. A dedicated worker pool configuration should look like the following:
+
+```tf
+cluster "packet" {
+  ...
+
+  worker_pool "storage" {
+    count = 3
+    node_type = "c2.medium.x86"
+
+    labels = "storage.lokomotive.io=ceph"
+    taints = "storage.lokomotive.io=ceph:NoSchedule"
+  }
+}
+```
+
+- The number of machines provided using `count` should be an odd number greater than or equal to three.
+- Type of node, provided using `node_type`, should be one that has multiple disks like `c2.medium.x86` or `s1.large.x86`. Find out more servers [here](https://www.packet.com/cloud/servers/).
+- To steer `rook-ceph` workload on these storage nodes provide `labels`.
+- Provide `taints` so that other workload can be **steered away** by default. This setting is not mandatory, but isolating storage workloads from others is recommended so that other workloads do not degrade the performance of the storage.
+
+#### Deploy the worker pool
+
+Execute the following command to deploy the `storage` worker pool:
+
+```bash
+lokoctl cluster apply -v --skip-components
+```
+
+## Deploy `rook`
+
+#### Config
+
+Create a file named `storage.lokocfg` with the following contents:
+
+```tf
+component "rook" {
+  node_selector = {
+    "storage.lokomotive.io" = "ceph"
+  }
+
+  toleration {
+    key      = "storage.lokomotive.io"
+    operator = "Equal"
+    value    = "ceph"
+    effect   = "NoSchedule"
+  }
+
+  agent_toleration_key    = "storage.lokomotive.io"
+  agent_toleration_effect = "NoSchedule"
+
+  discover_toleration_key    = "storage.lokomotive.io"
+  discover_toleration_effect = "NoSchedule"
+}
+```
+
+- `node_selector` should match the `labels` variable provided in the `worker_pool`.
```

Comment:
```suggestion
- `node_selector` should match the `labels` attribute provided in the `worker_pool`.
```

and everywhere variable is used. Just to distinguish between hcl variable and also maintain consistency with the configuration reference documents where the table of all attributes is called as `Attribute reference`.